### PR TITLE
Creating and Implemention Security HashString Method.

### PR DESCRIPTION
**Pull request by Maryann529**
Kindly review my code!

### CODE REVIEW

File: C-Sharp-Tutorial/C-Sharp-Tutorial.csproj [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (dabensymon) [inline C-Sharp-Tutorial/C-Sharp-Tutorial.csproj]: Why is this here? Please remove it.

File: C-Sharp-Tutorial/Program.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (dabensymon) [inline C-Sharp-Tutorial/Program.cs]: Same here. What is the purpose of this?

File: src/DwitTech.AccountService.Core/Interfaces/ISecurityRepository.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Interfaces/ISecurityRepository.cs]: What is this for? doesnt seem to have a purpose

File: src/DwitTech.AccountService.Core/Services/SecurityService.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            string collectedpassword = userpassword;
+            byte[] hashpassword;
+            byte[] asciibyte;
+
+            asciibyte= ASCIIEncoding.ASCII.GetBytes(collectedpassword);
+            hashpassword = new MD5CryptoServiceProvider().ComputeHash(asciibyte);
+            return ConvertByteToString(hashpassword);
+        }
+
+        /*public bool VerifiedHashPasswords(byte hashpassword1, byte hashpassword2)
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/SecurityService.cs]: Do not comment out code. Delete it instead.

File: src/DwitTech.AccountService.Core/Services/SecurityService.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+using System.Security.Cryptography;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.AccountService.Core.Services
+{
+    public class SecurityService
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/SecurityService.cs]: This should implement an interface.

File: src/DwitTech.AccountService.Core/Services/SecurityService.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        private string ConvertByteToString(byte[] hashpassword)
+        {
+            StringBuilder hashedstringoutput = new StringBuilder(hashpassword.Length);
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/SecurityService.cs]: variable and parameter names should be camelCase. All your variables and parameters are guilty of this.

**Review by dabensymon — CHANGES_REQUESTED** [review]

File: src/DwitTech.AccountService.Core/Services/SecurityService.cs [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.AccountService.Core.Services
+{
+    public class SecurityService : ISecurityService
+    {
+        
+        public string HashString(string userPassword)
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/SecurityService.cs]: This method hashes any string, not just password. So it should be clearText or inputString and not userPassword

File: src/DwitTech.AccountService.Core/Services/SecurityService.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            if (userPassword == null)
+            {
+                return null;
+            }
+            string collectedPassword = userPassword;
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/SecurityService.cs]: why are you assigning another variable here?

File: src/DwitTech.AccountService.Core/Services/SecurityService.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+            byte[] hashPassword;
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/SecurityService.cs]: remove all mention of password from this implementation cos its not just a password thing

File: src/DwitTech.AccountService.Core/Services/SecurityService.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+            byte[] asciibyte;
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/SecurityService.cs]: camelCase please

File: src/DwitTech.AccountService.Core/Services/SecurityService.cs [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+            asciibyte = ASCIIEncoding.ASCII.GetBytes(collectedPassword);
+            hashPassword = new MD5CryptoServiceProvider().ComputeHash(asciibyte);
+            return ConvertByteToString(hashPassword);
+        }
+
+         private string ConvertByteToString(byte[] hashPassword)
+        {
+            StringBuilder hashedStringOutput = new StringBuilder(hashPassword.Length);
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/SecurityService.cs]: bad naming here again. This should be called something like stringBuilder or outputStringBuilder

**Review by dabensymon — COMMENTED** [review]
Not bad. Just fix the few comments lets close it.

**Review by dabensymon — COMMENTED** [review]
Still pending

**Review by giftedGeorge — DISMISSED** [review]
Looks good to me

**Review by giftedGeorge — COMMENTED** [review]
Looks good to me

File: src/DwitTech.AccountService.Core/Services/SecurityService.cs [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+    public class SecurityService : ISecurityService
+    {
+        
+        public string HashString(string inputString)
+        {
+            if (inputString == null)
+            {
+                return null;
+            }
+            /*string collectedPassword = inputString;*/
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/SecurityService.cs]: why is this commented out code still here?

**Review by dabensymon — COMMENTED** [review]
Just 1 issue

**Review by dabensymon — APPROVED** [review]

**Pull request merged**